GNUmakefile

```diff
@@ -2,6 +2,29 @@ MAKEFLAGS += --jobs=1
 
 -include config.unix
 
+#Lua version detection
+
+ifeq ($(origin LUA_VERSION),undefined)
+  # Get Lua version directly from the LUA interpreter 
+  ifneq ($(origin LUA),undefined)
+    LUA_VERSION := $(shell $(LUA) -e "print((_VERSION or ''):match('Lua (%d+%.%d+)') or '')")
+  else
+    
+	# system lua if LUA isn't defined
```

Comment:
This should probably be consolidated by just setting `LUA ?= lua`. That way you don't have to check if the value is set and have two different control structures, you can just do one pass.